src/YesSql.Core/Session.cs

```diff
@@ -240,6 +240,31 @@ public void Detach(IEnumerable<object> entries, string collection)
             }
         }
 
+        public void DetachAll(string collection)
+        {
+            CheckDisposed();
+
+            var state = GetState(collection);
+
+            foreach (var item in state.IdentityMap.GetAll())
+            {
+                state.Saved.Remove(item);
+                state.Updated.Remove(item);
+                state.Tracked.Remove(item);
+                state.Deleted.Remove(item);
```

Comment:
```
      state._concurrent?.Clear();
      state._saved?.Clear();
      state._updated?.Clear();
      state._tracked?.Clear();
      state._deleted?.Clear();
```

might allocate less, as the state may not have created those hashsets, i.e. you might be creating them just by trying to remove an item.

but minor.